compiler/rustc_codegen_gcc/src/back/write.rs

```diff
@@ -113,6 +125,9 @@ pub(crate) unsafe fn codegen(
                     context.set_debug_info(true);
                     context.dump_to_file(path, true);
                 }
+                if config.json_artifact_notifications {
+                    dcx.emit_artifact_notification(&obj_out, "obj");
+                }
```

Comment:
This should be moved after the `compile_to_file` calls. Otherwise the notification would be emitted without actually having written any object file.